### PR TITLE
Metrics webapp

### PR DESCRIPTION
**Pull request by AlyAtalla**
Use React documentation.
Use React components.
Use React props.
Use React Router.
Connect React and Redux.
Handle events in a React app.
Write integration tests with a React testing library.
Use styles in a React app.
Use React life cycle methods.
Apply React best practices and language style guides in code.
Use store, actions and reducers in React.

https://www.loom.com/share/8e525751157c4a78883d99057fc3cbe6

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,180 @@
...
+If you like this project give it a⭐️
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+- Microverse 
+- my coding partner Shah Zaman
```
Comment (ichala) [inline README.md:173]: - [x] as part of the requirements The README should contains an "Acknowledgements" section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/). kindly fix it 🙏

File: src/__tests__/Search.test.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        <Provider store={store}>
+          <Router>
+            <Search />
+          </Router>
+        </Provider>
+      </React.StrictMode>,
+    );
+    expect(tree).toMatchSnapshot();
+  });
+});
```
Comment (ichala) [inline src/__tests__/Search.test.js]: - [ ] good job testing your components 👏 but as part of the requirements kindly add some unit & integration tests for some pure functions in your app 🙏

File: src/Components/Home.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      </div>
+      <div className="testing">
+        <div className="marvel-characters">
+          <div className="character-title">
+            <Search search={(q) => setQuery(q)} />
+          </div>
+          {heros.length ? heros.map((hero) => <Link state={hero} className="characters-link" key={`heroid-${hero.id}`} to="Comics"><Character newhero={hero} /></Link>) : <div className="loading">Loading...</div>}
+        </div>
+      </div>
+    </div>
```
Comment (ichala) [inline src/Components/Home.js:39]: - [ ] while your design look stunning and awesome you should stick to the design guidelines as part of the requirements , including : 
1. Typography: font face, size, and weight
   * Layout: composition and space between elements.
1. You should select one main colour for the app and be consistent (it does **not** need to be the same colour as in the design guidelines).

Your Design           | Guidelines 
:-------------------------:|:-------------------------:
image | image

**Review by ichala — CHANGES_REQUESTED** [review]
Hi Student
Good job so far! 
👏👏👏 
You still need to work on some issues to go to the next project, but you're almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- No Linter errors  ✔️ 
- The project uses Redux for state management and data retrieved from the API are stored in the Redux store correctly✔️
- The project contains a `Home` page and a `Detail` page✔️
- It's possible to navigate the detail page by clicking the individual category item✔️
- The project is a single page application (SPA) using `react-router-dom` to serve each page without reloading✔️


 ## Required Changes ♻️

- [ ] **_🛑 As part of the requirements, you should deploy the application using Render, Netlify or GitHub Pages , and make sure to add the live link in the readme file 🙏_**
- [ ] **_🛑  As part of the requirements, kindly record a presentation video where you presenting the features of the project and make sure that :_**
- The maximum length of the video is 5 minutes 
- The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear 
- The video contains a demo of the project, highlighting all the project features 
- The video contains a highlight about a piece of the code the student finds interesting or is proud of making
- [ ] **_🛑  Check The Comments Below_**





## Optional suggestions

- N/A

**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__tests__/Search.test.js (hunk 2 above)
Comment (AlyAtalla) [reply src/__tests__/Search.test.js]: thank you

File: README.md (hunk 1 above)
Comment (AlyAtalla) [reply README.md:173]: thank you

File: README.md [hunk 4]
```diff
@@ -1,2 +1,185 @@
...
+-credit to Nelson Sakwa the author of the original design
```
Comment (topeogunleye) [inline README.md:174]: - [ ] [OPTIONAL] I think it'll be better if you also link the benahance design to thIS TEXT". For Example:
`Original design idea by [Nelson Sakwa on Behance.](https://www.behance.net/gallery/31579789/Ballhead-App-%28Free-PSDs%29). Under the [Creative Commons License](https://creativecommons.org/licenses/by-nc/4.0/)`

**Review by topeogunleye — APPROVED** [review]
Hi @AlyAtalla, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

> ## (Highlights) Good Points: 💪 
> * Added unit & integration tests for some pure functions. :heavy_check_mark:
> * Added demo video. :heavy_check_mark:
> * Deployed live version on GitHub and added live version link to README file :heavy_check_mark:

çrpik-kurdî-impressive



Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear
Please, remember to tag @topeogunleye in your question so I can receive the notification.
 
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**